File: _scripts/function-directive.mjs
const functionDirective = {
  name: 'funky',
  doc: 'A directive for a displayed function.',
  alias: ['function'],
  arg: {
    type: 'myst',
  },
  options: {
    enumerated: {
      type: 'boolean',
      default: true,
      doc: 'Whether the function is enumerated.',
    },
  },
  body: {
    type: 'myst',
    required: true,
  },
  run(data) {
    const children = [];
    if (data.arg) {
      children.push({
        type: 'admonitionTitle',
        children: data.arg,
      });
    }
    if (data.body) {
      children.push(...(data.body));
    }
    enumerated = true;
    const func = {
      type: 'function',
      kind: 'function',
      enumerated,
      children: children,
    };
    return [func];
  },
};

const plugin = { name: 'Function directive', directives: [functionDirective] };

export default plugin;

File: _scripts/function-directive.py
#!/usr/bin/env python3
import argparse
import json
import sys
import re


plugin = {
    "name": "Function Directive",
    "directives": [
        {
            "name": "function",
            "doc": "",
            "alias": ["function"],
            "arg": {
                "type": "myst",
            },
            "options": {
                
            },
        }
    ],
}


def declare_result(content):
    """Declare result as JSON to stdout

    :param content: content to declare as the result
    """

    # Format result and write to stdout
    json.dump(content, sys.stdout, indent=2)
    # Successfully exit
    raise SystemExit(0)


def run_directive(name, data):
    """Execute a directive with the given name and data

    :param name: name of the directive to run
    :param data: data of the directive to run
    """
    assert name == "function"

    name = data.get("arg")

    # Insert an image of a landscape
    func = []
    func.append({
        "type": "admonitionTitle",
        "children": name,
    })
    func.extend(data.get("body"))
    return func


if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    group = parser.add_mutually_exclusive_group()
    group.add_argument("--role")
    group.add_argument("--directive")
    group.add_argument("--transform")
    args = parser.parse_args()

    if args.directive:
        data = json.load(sys.stdin)
        declare_result(run_directive(args.directive, data))
    elif args.transform:
        raise NotImplementedError
    elif args.role:
        raise NotImplementedError
    else:
        declare_result(plugin)
